ow_dynamic_terrain/src/DynamicTerrainBase.cpp

```diff
@@ -0,0 +1,68 @@
+#include "DynamicTerrainBase.h"
+#include "memory_ext.h"
+
+using namespace std;
+using namespace ow_dynamic_terrain;
+
+void DynamicTerrainBase::Initialize(const std::string& topic_extension)
+{
+  if (!ros::isInitialized())
+  {
+    gzerr << m_plugin_name + ": ROS not initilized! The plugin won't load" << endl;
+    return;
+  }
+
+  m_node_handle = make_unique<ros::NodeHandle>(m_plugin_name);
+  m_node_handle->setCallbackQueue(&m_callback_queue);
+
+  auto on_modify_terrain_circle = [this](const modify_terrain_circle::ConstPtr& msg) {
```

Comment:
This is a good use for auto.

Reply:
I don't agree to this narrative. It is almost always good to use **auto**. In any case, I have amended the PR to declare the type of each variable as I don't wish to delay the PR any further.